syntax = "proto3";
option cc_generic_services = true;
option java_generic_services = true;

import "common.proto";

package actions; 

enum ActionType
{
   Task = 0;
   Group = 1;
}

message ActionInfo
{
   common.UniqueId id = 1;
   string title = 3;
   string description = 4;
   ActionType type = 5;
   common.UniqueId data_source_id = 6;
}

message ActionsList
{
   repeated ActionInfo list = 1;
}

service ActionsService 
{
  rpc GetItems (common.EmptyMessage) returns (ActionsList);
  
  rpc AddElement (ActionInfo) returns (common.UniqueId);
  rpc DeleteElement (common.UniqueId) returns (common.OperationResultMessage);
  rpc EditElement (ActionInfo) returns (common.OperationResultMessage);
  rpc Clear(common.EmptyMessage) returns (common.OperationResultMessage);
}
